pyvirtualcam/native_windows_unity_capture/shared_memory/shared.inl

```diff
@@ -58,7 +58,7 @@ struct SharedImageMemory
 		callback(m_pSharedBuf->width, m_pSharedBuf->height, m_pSharedBuf->stride, (EFormat)m_pSharedBuf->format, (EResizeMode)m_pSharedBuf->resizemode, (EMirrorMode)m_pSharedBuf->mirrormode, m_pSharedBuf->timeout, m_pSharedBuf->data, callback_data);
 		ReleaseMutex(m_hMutex); //unlock mutex
 
-		return RECEIVERES_NEWFRAME;
+		return (IsNewFrame ? RECEIVERES_NEWFRAME : RECEIVERES_OLDFRAME);
```

Comment:
With this line, the timeout should be set to a large enough number (previously I set it to 100000000) in milliseconds or the frame will be cleared immediately.

Reply:
I just don't want the frame being cleared even after a day, but that's up to you.

Reply:
This is the receiver, so there's no point changing any code in it as it won't be used by pyvirtualcam but rather by the filter DLL that you install. I'll have a look at the timeout value on the sender side, it's a fair point.

Reply:
Yeah.. the timeout definitely needs to be higher, otherwise low framerates also lead to interruptions. I've changed it to 1 day.

Reply:
The receive method is used [here](https://github.com/schellingb/UnityCapture/blob/fe461e8f6e1cd1e6a0dfa9891147c8e393a20a2c/Source/UnityCaptureFilter.cpp#L141) and this is the only place which prints "Unity has stopped sending image data". That's OK if you think 1 day is enough (though I want to keep the frame forever).

Reply:
The receive method is only used in the filter. I can remove the receive method in the backend and everything would compile fine. Do you see? Any change in that method does not have an effect on our side. The shared.inl is a header which gets pulled in both by the receiver (filter) and sender (us). Some code is only used by one of both.

What's your use case for keeping a frame longer than a day? I'm happy to make the timeout longer but would like to understand a bit more.

Reply:
The timeout is of type `int` (signed 32 bit) and since it's in milliseconds the maximum you can get is something like 24 days.

Reply:
I changed it to the maximum now :)
https://github.com/letmaik/pyvirtualcam/commit/d03b9a698e560dc9aafe6f555a4787f8307feedb

Reply:
Previously I don't see the commit and think that it would overflow.